spring-kafka/src/main/java/org/springframework/kafka/support/DefaultKafkaHeaderMapper.java

```diff
@@ -152,6 +156,9 @@ public DefaultKafkaHeaderMapper(ObjectMapper objectMapper, String... patterns) {
 		for (String pattern : patterns) {
 			this.matchers.add(new SimplePatternBasedHeaderMatcher(pattern));
 		}
+		@SuppressWarnings({ "rawtypes", "unchecked" })
+		Module module = new SimpleModule().addDeserializer(MimeType.class, new MimeTypeJsonDeserializer(objectMapper));
+		this.objectMapper.registerModule(module);
```

Comment:
That's cool of course!
But is it OK to mutate externally provided `ObjectMapper`? 
Especially in Spring Boot app when too much is based on globally auto-configured bean...

I won't mind though if we just document this with the IMPORTANT mark.